mcp/sensors/star_camera_receive.c

```diff
@@ -0,0 +1,347 @@
+/***************************************************************************
+ mcp: the TIM master control program
+ 
+ This software is copyright (C) 2002-2006 University of Toronto
+ 
+ This file is part of mcp.
+ 
+ mcp is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ at your option) any later version.
+ 
+ mcp is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+ 
+ You should have received a copy of the GNU General Public License
+ along with mcp; if not, write to the Free Software Foundation, Inc.,
+ 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ 
+ created by Ian Lowe 4-25-23
+ **************************************************************************/
+
+
+/*************************************************************************
+ 
+ star_camera_receive.c -- mcp code to connect to data streams from SC1 and SC2
+ 
+ *************************************************************************/
+
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <signal.h>
+#include <errno.h>
+
+#include "star_camera_structs.h"
+#include "star_camera_receive.h"
+#include "command_struct.h"
+
+// a lot of boilerplate stuff to set up the UDP sockets in a minimal way
+
```

Comment:
Only one function defined in this file is declared in the corresponding header. You've made it clear that the other functions are only for use here by declaring them all `static`, which is the right thing to do at this moment in time, because they really aren't for use by another consumer. I don't think really **should** be true forever, though. If you ever want to **unit test** any of these functions, it's good to be able to `#include` the header in the test .c file to call these functions for testing. 

If I was really being a stickler, I would require tests with this PR, and you'd quickly find out that this action would be required with that work. I'll leave it up to you if you want to leave things like this for now.

Reply:
Per discussion in slack: doing this results in a main code polluted namespace with functions that truly do not need to be seen anywhere else. We can get around this in a slightly bad-form manner by including the .c files in the tests, which allows access to the functions while preventing namespace pollution in the main software.